docker-app/qfieldcloud/core/drf_utils.py

```diff
@@ -0,0 +1,124 @@
+    It is possible to use an `ordering_fields` expression value with attributes.
+    Custom attributes expression has form : "my_field::alias=my_field_alias,key=value"
+    """
+
+    SEPARATOR = "::"
+    TOKENS_LIST_SEPARATOR = ","
+    TOKENS_VALUE_SEPARATOR = "="
+
+    def _get_query_field(self, fields: list[str], term: str) -> str | None:
+        """Searches a term in a query field list.
+
+        The field list elements may start with "-".
+        This method should be used to search a term from a query's ordering fields.
+
+        Args:
+            fields (list[str]): list of fields to search
+            term (str): term to search in the list
+        Returns:
+            str | None: the matching field, if present in the list
+        """
+        for field in fields:
+            compare_value = field
+
+            if field.startswith("-"):
+                compare_value = field[1:]
+
+            if compare_value != term:
+                continue
+
+            return field
+        return None
+
+    def _parse_tokenized_attributes(self, raw: str) -> dict[str, str]:
+        """Parses an ordering field attributes expression.
+
+        Args:
+            raw (str): raw expression to parse, e.g.: "alias=my_field_alias,key=value"
+        Returns:
+            dict[str, str]: dict containing the expression's attributes
+        """
+        definition_attrs = raw.split(self.TOKENS_LIST_SEPARATOR)
+
+        attr_dict = {}
+        for attr in definition_attrs:
+            token, value = attr.split(self.TOKENS_VALUE_SEPARATOR, 1)
+            attr_dict[token] = value
+
+        return attr_dict
+
+    def _parse_definition(self, definition: str) -> tuple[str, dict[str, str]]:
+        """Parses a custom ordering field with attributes expression.
+
+        Args:
+            definition (str): raw definition of the ordering field to parse,
+                e.g.: "my_field::alias=my_field_alias,key=value"
+        Returns :
+            tuple[str, dict[str, str]]: tuple containing the field name (1st) and its attributes dict (2nd)
+        """
+        name, attr_str = definition.split(self.SEPARATOR, 1)
+        attrs = self._parse_tokenized_attributes(attr_str)
+
+        return name, attrs
+
+    def remove_invalid_fields(
+        self,
+        queryset: QuerySet,
+        fields: list[str],
+        view: viewsets.ModelViewSet,
+        request: HttpRequest,
+    ) -> list[str]:
+        """Process ordering fields by parsing custom field expression.
+
+        Custom attributes expression has form : "my_field::alias=my_field_alias,key=value".
+        In the above example, `alias` is the URL GET param value,
+        but `my_field` is the real model field.
+
+        Args:
+            queryset (QuerySet): Django's Project queryset
+            fields (list[str]): HTTP query's ordering fields
+            view (ModelViewSet): DRF's Project view
+            request (HttpRequest): incoming HTTP request
```

Comment:
```suggestion
            queryset (QuerySet): Django's ORM queryset
            fields (list[str]): ordering fields passed to the HTTP querystring
            view (ModelViewSet): DRF view instance
            request (HttpRequest): DRF request instance
```

Note that the Project queryset and the Project view are a special usecase, the Filter should be generic.